service/storage/operations.go

```diff
@@ -287,17 +287,23 @@ func (l *Library) IterateLedger(callback func(path ledger.Path, payload *ledger.
 		it := tx.NewIterator(opts)
 		defer it.Close()
 
-		next := encodeKey(prefixPayload, highest, uint64(math.MaxUint64))
-		for it.Seek(next); it.ValidForPrefix(prefix); it.Seek(next) {
+		sentinel := encodeKey(prefixPayload, highest, uint64(math.MaxUint64))
+		for it.Seek(sentinel); it.ValidForPrefix(prefix); it.Next() {
```

Comment:
This should be 
```suggestion
		for it.Seek(sentinel); it.ValidForPrefix(prefix); it.Seek(sentinel) {
```

and the `Seek()` at the bottom of the loop should go, right?

When I tried it locally with the "static" indexer, that worked, while the current version did not.

Reply:
Correction - I think the loop should be:

```go
for it.Seek(sentinel); it.ValidForPrefix(prefix); {
    // stuff
	height := binary.BigEndian.Uint64(key[33:41])
	if exclude(height) {
		it.Next()
		continue
	}

	// bottom of the loop
	sentinel = encodeKey(prefixPayload, path, uint64(math.MaxUint64))
	it.Seek(sentinel)
}

```

Something like this worked for me both for resuming from checkpoint and resuming from index